### ISSUE TITLE
Bottom edge of Tabman bar is not pinned to the Bottom edge of the safe area in iPhone X.

### ISSUE THREAD

**Issue by annjawn** [uias/Tabman#154]
With iOS 11, Xcode 9 and iPhone X Apple introduced a concept of "Safe Area" in a UIViewController. I have implemented Tabman in my project and it works great in all the devices except for iPhone X where the Tab seems to hover above the Safe Area (I guess) which gives it a weird effect. How does one go about fixing this?

![image](https://user-images.githubusercontent.com/26977294/31322184-c6c3ba2e-ac47-11e7-94e0-2913a5a76ff6.png)

So, it turns out it's a pretty big bummer. Apprently, this method in the Tabman+Layout.swift file will need to be modified to use Apple's new "Safe Area Layout Guide", but this method returns constraints using the visual format language. However, according to reports, the visual format language is not available for "Safe Area Layout Guide" yet and it's perhaps a bug. 

```
@discardableResult func barAutoPinToBotton(bottomLayoutGuide: UILayoutSupport) -> [NSLayoutConstraint]? {
        guard self.superview != nil else {
            return nil
        }
        self.translatesAutoresizingMaskIntoConstraints = false

        var constraints = [NSLayoutConstraint]()
        
        let margins = self.layoutMargins
        let views: [String: Any] = ["view" : self, "bottomLayoutGuide" : bottomLayoutGuide]
        let xConstraints = NSLayoutConstraint.constraints(withVisualFormat: "H:|-0-[view]-0-|",
                                                          options: NSLayoutFormatOptions(),
                                                          metrics: nil, views: views)
        let yConstraints = NSLayoutConstraint.constraints(withVisualFormat: String(format: "V:[view]-%i-[bottomLayoutGuide]", -margins.bottom),
                                                          options: NSLayoutFormatOptions(),
                                                          metrics: nil, views: views)
        constraints.append(contentsOf: xConstraints)
        constraints.append(contentsOf: yConstraints)
        
        self.superview?.addConstraints(constraints)
        return constraints
    }
```

**Comment by msaps** [comment]
@annjawn I think it's more of an issue with the background not extending over the function area on the iPhone X. Looks as though the bar is positionally where it should be, but the background should extend below (as `UITabBar` does). 

I will also update from any usage of `bottomLayoutGuide` / `topLayoutGuide` to use the safe area insets. `1.0.2` addressed quite a bit of this in regards to automatic child insetting, but there are obviously still parts that need updating!

Will try and get it sorted ASAP 😄

**Comment by annjawn** [comment]
@msaps technically the workaround is silly and simple and would be applicable to people migrating their projects from Xcode 8 to 9. In Xcode 9 there is an option to update your story boards to use "Safe area" so the storyboard documents can be updated to use that option (Xcode 9 will NOT update your storyboard to use safe area for your migrated projects) so it will have to be done manually and then the contents of the view controller will need to be pinned to the bottom edge of the Safe area instead of Bottom edge of Superview but that could mean fixing other constraints on the view controller too.

![image](https://user-images.githubusercontent.com/26977294/31409610-5ef45dc4-adc1-11e7-987f-6398f30a80b0.png)

![image](https://user-images.githubusercontent.com/26977294/31412055-7f7397ac-adc8-11e7-8ac0-3c6523d08065.png)

**Comment by msaps** [comment]
This should now be resolved as of [1.0.3](https://github.com/uias/Tabman/releases/tag/1.0.3). 😄 

You need to have `extendBackgroundEdgeInsets` enabled in `TabmanBar.Appearance.Layout` for this to work (It's enabled by default). 

But should give you something that looks like this...

<img width="444" alt="screen shot 2017-10-11 at 20 33 34" src="https://user-images.githubusercontent.com/3515448/31462950-a2a09764-aece-11e7-86d7-25621ac9a183.png">

**Comment by annjawn** [comment]
Nice!
